src/vdom/component.js

```diff
@@ -33,26 +33,18 @@ export function setComponentProps(component, props, renderMode, context, mountAl
 		}
 	}
 
-	if (context && context!==component.context) {
-		if (!component.prevContext) component.prevContext = component.context;
-		component.context = context;
-	}
-
-	if (!component.prevProps) component.prevProps = component.props;
-	component.props = props;
-
 	component._disable = false;
 
 	if (renderMode!==NO_RENDER) {
 		if (renderMode===SYNC_RENDER || options.syncComponentUpdates!==false || !component.base) {
-			renderComponent(component, SYNC_RENDER, mountAll);
+			renderComponent(component, props, context, SYNC_RENDER, mountAll);
 		}
 		else {
-			enqueueRender(component);
+			enqueueRender(component, props, context);
 		}
 	}
 
-	applyRef(component.__ref, component);
+	applyRef(component.__ref, component); // kvndy TODO: props and context may not be correct here
```

Comment:
With this PR, props and context would only be set in renderComponent. I find it fortuitous that the refactoring led me to the same conclusion as this comment: https://github.com/developit/preact/issues/1177#issuecomment-416054414 I haven't yet had to make any "decisions" so to speak, and seek recommendation.